diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,10 +88,12 @@
 if (SIRIT_USE_SYSTEM_SPIRV_HEADERS)
     find_package(SPIRV-Headers REQUIRED)
 else()
-    if (NOT TARGET SPIRV-Headers)
+    if (NOT TARGET SPIRV-Headers AND NOT TARGET SPIRV-Headers::SPIRV-Headers)
         add_subdirectory(externals/SPIRV-Headers EXCLUDE_FROM_ALL)
-        add_library(SPIRV-Headers::SPIRV-Headers ALIAS SPIRV-Headers)
+    endif()
+    if (TARGET SPIRV-Headers AND NOT TARGET SPIRV-Headers::SPIRV-Headers)
+        add_library(SPIRV-Headers::SPIRV-Headers ALIAS SPIRV-Headers)
     endif()
 endif()
